pkg/providers/ovirt/client/ovirt-client.go

```diff
@@ -0,0 +1,272 @@
+package ovirtclient
+
+import (
+	"fmt"
+
+	ovirtsdk "github.com/ovirt/go-ovirt"
+)
+
+// OvirtClient retrieves rich VM object - with all necessary links have been followed
+type OvirtClient interface {
+	GetVM(id *string, name *string, cluster *string) (*ovirtsdk.Vm, error)
+	Close() error
+}
+
+// ConnectionSettings wrap information required to make oVirt API connection
+type ConnectionSettings struct {
+	URL      string
+	Username string
+	Password string
+	Insecure bool
+	CAFile   string
+}
+
+// RichOvirtClient is responsilbe for retrieving VM data from oVirt API
+type richOvirtClient struct {
+	connection *ovirtsdk.Connection
+}
+
+// NewRitchOvirtClient creates new, connected rich oVirt client. After it is no longer needed, call Close().
+func NewRitchOvirtClient(cs *ConnectionSettings) (OvirtClient, error) {
+	con, err := connect(cs.URL, cs.Username, cs.Password, cs.CAFile, cs.Insecure)
+	if err != nil {
+		return nil, err
+	}
+	client := richOvirtClient{
+		connection: con,
+	}
+	return &client, nil
+}
+
+// Close releases the resources used by this client.
+func (client *richOvirtClient) Close() error {
+	return client.connection.Close()
+}
+
+// GetVM rertrieves oVirt VM data for given id or name and cluster. VM will have certain links followed and updated.
+func (client *richOvirtClient) GetVM(id *string, name *string, cluster *string) (*ovirtsdk.Vm, error) {
```

Comment:
Note that we'll have to change the method to support cluster identification by name or id after the CRD patch is merged:
https://github.com/kubevirt/vm-import-operator/pull/9/files#diff-6c071afa80342d59a0ce5806f5daf604R77

Reply:
Thanks for pointing that out!